diff -ruN src/SuiteSparse/SuiteSparse_config/Makefile src/SuiteSparse-patched/SuiteSparse_config/Makefile
--- src/SuiteSparse/SuiteSparse_config/Makefile	1969-12-31 19:00:00
+++ src/SuiteSparse-patched/SuiteSparse_config/Makefile	2024-01-12 11:54:13
@@ -0,0 +1,16 @@
+sources = SuiteSparse_config.c
+objects = $(sources:.c=.o)
+archive = SuiteSparse_config.a
+
+all : $(archive)
+
+$(archive) : $(objects)
+	rm -f $@
+	$(AR) -cr $@ $(objects)
+	$(RANLIB) $@
+
+.c.o :
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $*.o
+
+clean :
+	@rm -f $(objects) $(archive)
diff -ruN src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.c src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.c
--- src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.c	2024-01-10 19:08:51
+++ src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.c	2024-01-12 11:54:13
@@ -72,8 +72,8 @@
             /* MATLAB mexFunction: */
             mexPrintf,
         #else
-            /* standard ANSI C: */
-            printf,
+#include <R_ext/Print.h>
+            Rprintf,
         #endif
     #else
         /* printf is disabled */
diff -ruN src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.h src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.h
--- src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.h	2024-01-10 19:08:51
+++ src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.h	2024-01-12 11:54:13
@@ -678,6 +678,19 @@
 
 #if defined ( SUITESPARSE_BLAS_DEFINITIONS )
 
+#ifndef USE_FC_LEN_T
+# define USE_FC_LEN_T
+#endif
+#include <Rconfig.h>
+
+#ifdef FC_LEN_T
+# define FCLEN , FC_LEN_T
+# define FCONE , (FC_LEN_T) 1
+#else
+# define FCLEN
+# define FCONE
+#endif
+
 //------------------------------------------------------------------------------
 // gemv: Y = alpha*A*x + beta*Y
 //------------------------------------------------------------------------------
@@ -697,7 +710,7 @@
     // input/output:
     double *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -710,7 +723,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -729,7 +742,7 @@
     // input/output:
     float *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_sgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -742,7 +755,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_SGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -761,7 +774,7 @@
     // input/output:
     void *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -774,7 +787,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -793,7 +806,7 @@
     // input/output:
     void *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -806,7 +819,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -826,7 +839,7 @@
     // input/output:
     double *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dtrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -837,7 +850,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -853,7 +866,7 @@
     // input/output:
     float *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_strsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -864,7 +877,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_STRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -880,7 +893,7 @@
     // input/output:
     void *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ztrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -891,7 +904,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -907,7 +920,7 @@
     // input/output:
     void *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ctrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -918,7 +931,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -941,7 +954,7 @@
     // input/output:
     double *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dtrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -953,7 +966,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -972,7 +985,7 @@
     // input/output:
     float *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_strsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -984,7 +997,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_STRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1003,7 +1016,7 @@
     // input/output:
     void *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ztrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -1015,7 +1028,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1034,7 +1047,7 @@
     // input/output:
     void *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ctrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -1046,7 +1059,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1071,7 +1084,7 @@
     // input/output:
     double *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1087,7 +1100,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_DGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1108,7 +1121,7 @@
     // input/output:
     float *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_sgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1124,7 +1137,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_SGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1145,7 +1158,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1161,7 +1174,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_ZGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1182,7 +1195,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1198,7 +1211,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_CGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1220,7 +1233,7 @@
     // input/output:
     double *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dsyrk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1232,7 +1245,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DSYRK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1250,7 +1263,7 @@
     // input/output:
     float *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ssyrk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1262,7 +1275,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_SSYRK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1280,7 +1293,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zherk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1292,7 +1305,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZHERK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1310,7 +1323,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cherk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1322,7 +1335,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CHERK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1340,7 +1353,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1352,7 +1365,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_DPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-          &LAPACK_Info) ;                                                     \
+          &LAPACK_Info FCONE) ;                                               \
         info = (Int) LAPACK_Info ;                                            \
     }                                                                         \
 }
@@ -1367,7 +1380,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_spotrf(uplo,n,A,lda,info,ok)                       \
@@ -1379,7 +1392,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_SPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-          &LAPACK_Info) ;                                                     \
+          &LAPACK_Info FCONE) ;                                               \
         info = (Int) LAPACK_Info ;                                            \
     }                                                                         \
 }
@@ -1394,7 +1407,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1406,7 +1419,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_ZPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-            &LAPACK_Info) ;                                                   \
+            &LAPACK_Info FCONE) ;                                             \
         info = LAPACK_Info ;                                                  \
     }                                                                         \
 }
@@ -1421,7 +1434,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_cpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1433,7 +1446,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_CPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-            &LAPACK_Info) ;                                                   \
+            &LAPACK_Info FCONE) ;                                             \
         info = LAPACK_Info ;                                                  \
     }                                                                         \
 }
@@ -1667,7 +1680,7 @@
     // output:
     double *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1679,7 +1692,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_DLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1696,7 +1709,7 @@
     // output:
     float *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1708,7 +1721,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_SLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1725,7 +1738,7 @@
     // output:
     void *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1737,7 +1750,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1754,7 +1767,7 @@
     // output:
     void *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1766,7 +1779,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_CLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1795,7 +1808,7 @@
     // workspace:
     double *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1812,7 +1825,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_DLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1837,7 +1850,7 @@
     // workspace:
     float *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1854,7 +1867,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_SLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1879,7 +1892,7 @@
     // workspace:
     void *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1896,7 +1909,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1921,7 +1934,7 @@
     // workspace:
     void *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1938,7 +1951,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_CLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -2140,7 +2153,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    double *Work
+    double *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2152,7 +2165,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_DLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2170,7 +2183,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    float *Work
+    float *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2182,7 +2195,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_SLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2200,7 +2213,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    void *Work
+    void *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2212,7 +2225,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2230,7 +2243,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    void *Work
+    void *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2242,7 +2255,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_CLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
