--- AMD/Include/amd_internal.h~	2010-07-25 04:32:06.000000000 +0200
+++ AMD/Include/amd_internal.h	2012-01-13 10:08:23.001072000 +0100
@@ -27,6 +27,8 @@
  *	    function pointers amd_malloc, amd_free, amd_realloc, and
  *	    amd_calloc at run-time for AMD to work properly.
  */
+// For use with R package 'Matrix':
+#define NPRINT
 
 /* ========================================================================= */
 /* === NDEBUG ============================================================== */
