--- CHOLMOD/Include/cholmod_core.h.~1~	2020-04-30 13:44:44.974447000 +0200
+++ CHOLMOD/Include/cholmod_core.h	2023-03-20 16:25:47.315469000 +0100
@@ -1579,8 +1579,9 @@
     cholmod_common *Common
 ) ;
 
-cholmod_sparse *cholmod_l_add (cholmod_sparse *, cholmod_sparse *, double *,
-    double *, int, int, cholmod_common *) ;
+cholmod_sparse *cholmod_l_add (cholmod_sparse *, cholmod_sparse *,
+			       double[2], double[2],
+			       int, int, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
 /* cholmod_sparse_xtype: change the xtype of a sparse matrix */
