diff -ubBw SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_cholesky.h CHOLMOD/Include/cholmod_cholesky.h
--- SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_cholesky.h	2021-05-18 15:01:16.000000000 +0200
+++ CHOLMOD/Include/cholmod_cholesky.h	2023-03-21 11:15:41.009409000 +0100
@@ -165,7 +165,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_factorize_p (cholmod_sparse *, double *, SuiteSparse_long *,
+int cholmod_l_factorize_p (cholmod_sparse *, double [2], SuiteSparse_long *,
     size_t, cholmod_factor *, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
@@ -384,7 +384,8 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_rowfac (cholmod_sparse *, cholmod_sparse *, double *, size_t,
+
+int cholmod_l_rowfac (cholmod_sparse *, cholmod_sparse *, double [2], size_t,
     size_t, cholmod_factor *, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
@@ -410,7 +411,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_rowfac_mask (cholmod_sparse *, cholmod_sparse *, double *, size_t,
+int cholmod_l_rowfac_mask (cholmod_sparse *, cholmod_sparse *, double [2], size_t,
     size_t, SuiteSparse_long *, SuiteSparse_long *, cholmod_factor *,
     cholmod_common *) ;
 
@@ -431,7 +432,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_rowfac_mask2 (cholmod_sparse *, cholmod_sparse *, double *,
+int cholmod_l_rowfac_mask2 (cholmod_sparse *, cholmod_sparse *, double [2],
     size_t, size_t, SuiteSparse_long *, SuiteSparse_long, SuiteSparse_long *,
     cholmod_factor *, cholmod_common *) ;
 
diff -ubBw SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_core.h CHOLMOD/Include/cholmod_core.h
--- SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_core.h	2021-05-18 15:01:16.000000000 +0200
+++ CHOLMOD/Include/cholmod_core.h	2023-03-21 11:13:49.847141935 +0100
@@ -1579,8 +1579,9 @@
     cholmod_common *Common
 ) ;
 
-cholmod_sparse *cholmod_l_add (cholmod_sparse *, cholmod_sparse *, double *,
-    double *, int, int, cholmod_common *) ;
+cholmod_sparse *cholmod_l_add (cholmod_sparse *, cholmod_sparse *,
+			       double [2], double [2],
+			       int, int, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
 /* cholmod_sparse_xtype: change the xtype of a sparse matrix */
diff -ubBw SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_matrixops.h CHOLMOD/Include/cholmod_matrixops.h
--- SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_matrixops.h	2021-05-18 15:01:16.000000000 +0200
+++ CHOLMOD/Include/cholmod_matrixops.h	2023-03-21 11:03:00.364915659 +0100
@@ -140,7 +140,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_sdmult (cholmod_sparse *, int, double *, double *,
+int cholmod_l_sdmult (cholmod_sparse *, int, double [2], double [2],
     cholmod_dense *, cholmod_dense *Y, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
diff -ubBw SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_modify.h CHOLMOD/Include/cholmod_modify.h
--- SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_modify.h	2021-05-18 15:01:16.000000000 +0200
+++ CHOLMOD/Include/cholmod_modify.h	2023-03-21 11:04:01.759846120 +0100
@@ -212,7 +212,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_rowadd_solve (size_t, cholmod_sparse *, double *,
+int cholmod_l_rowadd_solve (size_t, cholmod_sparse *, double [2],
     cholmod_factor *, cholmod_dense *, cholmod_dense *, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
@@ -238,7 +238,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_rowadd_mark (size_t, cholmod_sparse *, double *,
+int cholmod_l_rowadd_mark (size_t, cholmod_sparse *, double [2],
     SuiteSparse_long *, cholmod_factor *, cholmod_dense *, cholmod_dense *,
     cholmod_common *) ;
 
@@ -291,7 +291,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_rowdel_solve (size_t, cholmod_sparse *, double *,
+int cholmod_l_rowdel_solve (size_t, cholmod_sparse *, double [2],
     cholmod_factor *, cholmod_dense *, cholmod_dense *, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
@@ -317,7 +317,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_rowdel_mark (size_t, cholmod_sparse *, double *,
+int cholmod_l_rowdel_mark (size_t, cholmod_sparse *, double [2],
     SuiteSparse_long *, cholmod_factor *, cholmod_dense *, cholmod_dense *,
     cholmod_common *) ;
 
diff -ubBw SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_supernodal.h CHOLMOD/Include/cholmod_supernodal.h
--- SuiteSparse-5.10.1/CHOLMOD/Include/cholmod_supernodal.h	2021-05-18 15:01:16.000000000 +0200
+++ CHOLMOD/Include/cholmod_supernodal.h	2023-03-21 11:04:38.563850200 +0100
@@ -117,7 +117,7 @@
     cholmod_common *Common
 ) ;
 
-int cholmod_l_super_numeric (cholmod_sparse *, cholmod_sparse *, double *,
+int cholmod_l_super_numeric (cholmod_sparse *, cholmod_sparse *, double [2],
     cholmod_factor *, cholmod_common *) ;
 
 /* -------------------------------------------------------------------------- */
