--- COLAMD/Source/colamd_global.c.~1~	2010-07-25 04:32:06.000000000 +0200
+++ COLAMD/Source/colamd_global.c	2012-01-13 12:12:17.001781000 +0100
@@ -16,6 +16,9 @@
 int (*colamd_printf) (const char *, ...) = mexPrintf ;
 #else
 #include <stdio.h>
-int (*colamd_printf) (const char *, ...) = printf ;
+// For use with R package 'Matrix':
+#include <R_ext/Print.h>
+#define printf Rprintf
+void (*colamd_printf) (const char *, ...) = printf ;
 #endif
 #else
