--- COLAMD/Source/colamd_global.c~	2010-07-25 04:32:06.000000000 +0200
+++ COLAMD/Source/colamd_global.c	2012-01-13 10:29:10.000519000 +0100
@@ -8,6 +8,10 @@
  * http://www.cise.ufl.edu/research/sparse
  * -------------------------------------------------------------------------- */
 
+// For use with R package 'Matrix':
+#include <R.h>
+#define printf Rprintf
+
 /* Global variables for COLAMD */
 
 #ifndef NPRINT
