diff -ruN src/SuiteSparse/SuiteSparse_config/Makefile src/SuiteSparse-patched/SuiteSparse_config/Makefile
--- src/SuiteSparse/SuiteSparse_config/Makefile	1969-12-31 19:00:00
+++ src/SuiteSparse-patched/SuiteSparse_config/Makefile	2024-01-02 16:26:01
@@ -0,0 +1,16 @@
+sources = SuiteSparse_config.c
+objects = $(sources:.c=.o)
+archive = SuiteSparse_config.a
+
+all : $(archive)
+
+$(archive) : $(objects)
+	rm -f $@
+	$(AR) -cr $@ $(objects)
+	$(RANLIB) $@
+
+.c.o :
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $*.o
+
+clean :
+	@rm -f $(objects) $(archive)
diff -ruN src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.c src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.c
--- src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.c	2024-01-02 16:25:19
+++ src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.c	2024-01-02 16:26:01
@@ -72,8 +72,8 @@
             /* MATLAB mexFunction: */
             mexPrintf,
         #else
-            /* standard ANSI C: */
-            printf,
+#include <R_ext/Print.h>
+            Rprintf,
         #endif
     #else
         /* printf is disabled */
diff -ruN src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.h src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.h
--- src/SuiteSparse/SuiteSparse_config/SuiteSparse_config.h	2024-01-02 16:25:19
+++ src/SuiteSparse-patched/SuiteSparse_config/SuiteSparse_config.h	2024-01-02 17:02:25
@@ -673,6 +673,19 @@
 
 #if defined ( SUITESPARSE_BLAS_DEFINITIONS )
 
+#ifndef USE_FC_LEN_T
+# define USE_FC_LEN_T
+#endif
+#include <Rconfig.h>
+
+#ifdef FC_LEN_T
+# define FCLEN , FC_LEN_T
+# define FCONE , (FC_LEN_T) 1
+#else
+# define FCLEN
+# define FCONE
+#endif
+
 //------------------------------------------------------------------------------
 // gemv: Y = alpha*A*x + beta*Y
 //------------------------------------------------------------------------------
@@ -692,7 +705,7 @@
     // input/output:
     double *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -705,7 +718,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -724,7 +737,7 @@
     // input/output:
     float *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_sgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -737,7 +750,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_SGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -756,7 +769,7 @@
     // input/output:
     void *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -769,7 +782,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -788,7 +801,7 @@
     // input/output:
     void *Y,
     // input:
-    const SUITESPARSE_BLAS_INT *incy
+    const SUITESPARSE_BLAS_INT *incy FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cgemv(trans,m,n,alpha,A,lda,X,incx,beta,Y,incy,ok)   \
@@ -801,7 +814,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CGEMV (trans, &M_blas_int, &N_blas_int, alpha, A,    \
-            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int) ;      \
+            &LDA_blas_int, X, &INCX_blas_int, beta, Y, &INCY_blas_int FCONE) ; \
     }                                                                         \
 }
 
@@ -821,7 +834,7 @@
     // input/output:
     double *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dtrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -832,7 +845,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -848,7 +861,7 @@
     // input/output:
     float *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_strsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -859,7 +872,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_STRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -875,7 +888,7 @@
     // input/output:
     void *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ztrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -886,7 +899,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -902,7 +915,7 @@
     // input/output:
     void *X,
     // input:
-    const SUITESPARSE_BLAS_INT *incx
+    const SUITESPARSE_BLAS_INT *incx FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ctrsv(uplo,trans,diag,n,A,lda,X,incx,ok)             \
@@ -913,7 +926,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CTRSV (uplo, trans, diag, &N_blas_int, A,            \
-            &LDA_blas_int, X, &INCX_blas_int) ;                               \
+            &LDA_blas_int, X, &INCX_blas_int FCONE FCONE FCONE) ;             \
     }                                                                         \
 }
 
@@ -936,7 +949,7 @@
     // input/output:
     double *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dtrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -948,7 +961,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -967,7 +980,7 @@
     // input/output:
     float *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_strsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -979,7 +992,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_STRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -998,7 +1011,7 @@
     // input/output:
     void *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ztrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -1010,7 +1023,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1029,7 +1042,7 @@
     // input/output:
     void *B,
     // input:
-    const SUITESPARSE_BLAS_INT *ldb
+    const SUITESPARSE_BLAS_INT *ldb FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ctrsm(side,uplo,transa,diag,m,n,alpha,A,lda,B,ldb,ok)\
@@ -1041,7 +1054,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CTRSM (side, uplo, transa, diag, &M_blas_int,        \
-            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int) ;         \
+            &N_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int FCONE FCONE FCONE FCONE) ; \
     }                                                                         \
 }
 
@@ -1066,7 +1079,7 @@
     // input/output:
     double *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1082,7 +1095,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_DGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1103,7 +1116,7 @@
     // input/output:
     float *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_sgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1119,7 +1132,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_SGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1140,7 +1153,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1156,7 +1169,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_ZGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1177,7 +1190,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cgemm(transa,transb,m,n,k,alpha,A,lda,B,ldb,beta,    \
@@ -1193,7 +1206,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_CGEMM (transa, transb, &M_blas_int, &N_blas_int,     \
             &K_blas_int, alpha, A, &LDA_blas_int, B, &LDB_blas_int, beta, C,  \
-            &LDC_blas_int) ;                                                  \
+            &LDC_blas_int FCONE FCONE) ;                                      \
     }                                                                         \
 }
 
@@ -1215,7 +1228,7 @@
     // input/output:
     double *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_dsyrk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1227,7 +1240,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_DSYRK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1245,7 +1258,7 @@
     // input/output:
     float *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_ssyrk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1257,7 +1270,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_SSYRK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1275,7 +1288,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_zherk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1287,7 +1300,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_ZHERK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1305,7 +1318,7 @@
     // input/output:
     void *C,
     // input:
-    const SUITESPARSE_BLAS_INT *ldc
+    const SUITESPARSE_BLAS_INT *ldc FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_BLAS_cherk(uplo,trans,n,k,alpha,A,lda,beta,C,ldc,ok)      \
@@ -1317,7 +1330,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_BLAS_CHERK (uplo, trans, &N_blas_int, &K_blas_int, alpha, \
-            A, &LDA_blas_int, beta, C, &LDC_blas_int) ;                       \
+            A, &LDA_blas_int, beta, C, &LDC_blas_int FCONE FCONE) ;           \
     }                                                                         \
 }
 
@@ -1335,7 +1348,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1347,7 +1360,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_DPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-          &LAPACK_Info) ;                                                     \
+          &LAPACK_Info FCONE) ;                                               \
         info = (Int) LAPACK_Info ;                                            \
     }                                                                         \
 }
@@ -1362,7 +1375,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_spotrf(uplo,n,A,lda,info,ok)                       \
@@ -1374,7 +1387,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_SPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-          &LAPACK_Info) ;                                                     \
+          &LAPACK_Info FCONE) ;                                               \
         info = (Int) LAPACK_Info ;                                            \
     }                                                                         \
 }
@@ -1389,7 +1402,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1401,7 +1414,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_ZPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-            &LAPACK_Info) ;                                                   \
+            &LAPACK_Info FCONE) ;                                             \
         info = LAPACK_Info ;                                                  \
     }                                                                         \
 }
@@ -1416,7 +1429,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *lda,
     // output:
-    SUITESPARSE_BLAS_INT *info
+    SUITESPARSE_BLAS_INT *info FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_cpotrf(uplo,n,A,lda,info,ok)                       \
@@ -1428,7 +1441,7 @@
     {                                                                         \
         SUITESPARSE_BLAS_INT LAPACK_Info = -999 ;                             \
         SUITESPARSE_LAPACK_CPOTRF (uplo, &N_blas_int, A, &LDA_blas_int,       \
-            &LAPACK_Info) ;                                                   \
+            &LAPACK_Info FCONE) ;                                             \
         info = LAPACK_Info ;                                                  \
     }                                                                         \
 }
@@ -1662,7 +1675,7 @@
     // output:
     double *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1674,7 +1687,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_DLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1691,7 +1704,7 @@
     // output:
     float *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1703,7 +1716,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_SLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1720,7 +1733,7 @@
     // output:
     void *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1732,7 +1745,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1749,7 +1762,7 @@
     // output:
     void *T,
     // input:
-    const SUITESPARSE_BLAS_INT *ldt
+    const SUITESPARSE_BLAS_INT *ldt FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarft(direct,storev,n,k,V,ldv,Tau,T,ldt,ok)       \
@@ -1761,7 +1774,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_CLARFT (direct, storev, &N_blas_int, &K_blas_int,  \
-            V, &LDV_blas_int, Tau, T, &LDT_blas_int) ;                        \
+            V, &LDV_blas_int, Tau, T, &LDT_blas_int FCONE FCONE) ;            \
     }                                                                         \
 }
 
@@ -1790,7 +1803,7 @@
     // workspace:
     double *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1807,7 +1820,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_DLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1832,7 +1845,7 @@
     // workspace:
     float *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1849,7 +1862,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_SLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1874,7 +1887,7 @@
     // workspace:
     void *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1891,7 +1904,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -1916,7 +1929,7 @@
     // workspace:
     void *Work,
     // input:
-    const SUITESPARSE_BLAS_INT *ldwork
+    const SUITESPARSE_BLAS_INT *ldwork FCLEN FCLEN FCLEN FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarfb(side,trans,direct,storev,m,n,k,V,ldv,T,ldt, \
@@ -1933,7 +1946,7 @@
     {                                                                         \
         SUITESPARSE_LAPACK_CLARFB (side, trans, direct, storev, &M_blas_int,  \
             &N_blas_int, &K_blas_int, V, &LDV_blas_int, T, &LDT_blas_int, C,  \
-            &LDC_blas_int, Work, &LDWORK_blas_int) ;                          \
+            &LDC_blas_int, Work, &LDWORK_blas_int FCONE FCONE FCONE FCONE) ;  \
     }                                                                         \
 }
 
@@ -2135,7 +2148,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    double *Work
+    double *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_dlarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2147,7 +2160,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_DLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2165,7 +2178,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    float *Work
+    float *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_slarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2177,7 +2190,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_SLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2195,7 +2208,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    void *Work
+    void *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_zlarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2207,7 +2220,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_ZLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
@@ -2225,7 +2238,7 @@
     // input:
     const SUITESPARSE_BLAS_INT *ldc,
     // workspace:
-    void *Work
+    void *Work FCLEN
 ) ;
 
 #define SUITESPARSE_LAPACK_clarf(side,m,n,V,incv,tau,C,ldc,Work,ok)           \
@@ -2237,7 +2250,7 @@
     if (ok)                                                                   \
     {                                                                         \
         SUITESPARSE_LAPACK_CLARF (side, &M_blas_int, &N_blas_int, V,          \
-            &INCV_blas_int, tau, C, &LDC_blas_int, Work) ;                    \
+            &INCV_blas_int, tau, C, &LDC_blas_int, Work FCONE) ;              \
     }                                                                         \
 }
 
