--- src/CHOLMOD/MatrixOps/t_cholmod_sdmult.c~	2020-04-30 13:44:44.000000000 +0200
+++ src/CHOLMOD/MatrixOps/t_cholmod_sdmult.c	2022-08-30 21:16:38.028012000 +0200
@@ -79,8 +79,10 @@
     kcol = X->ncol ;
     dy = Y->d ;
     dx = X->d ;
-    w = W ;
-    Wz = W + 4*nx ;
+    if (A->stype != 0 && kcol >= 4) {
+	w = W ;
+	Wz = W + 4*nx ;
+    }
 
     /* ---------------------------------------------------------------------- */
     /* Y = beta * Y */
